common/lib/xmodule/xmodule/html_module.py

```diff
@@ -76,6 +76,17 @@ def student_view(self, _context):
         """
         return Fragment(self.get_html())
 
+    def student_view_data(self):
```

Comment:
Can you put this change behind a feature flag?  It's possible this change may bloat the size of our collected data - especially for some really large courses on edx.org.

FYI @ormsbee

Reply:
@nasthagiri
> Can you put this change behind a feature flag?
Do you want an instance-wide feature flag or one that's course specific?

Reply:
I believe an instance-wide flag (via a django setting) should be sufficient for this.  We may not enable it on edx.org, for instance, unless there's a need for it.

An alternative is to have a size-limit setting - this way, it is still safe to turn on - but with limits in place to protect against content explosion.

Reply:
Hi @nasthagiri 😄 
> An alternative is to have a size-limit setting - this way, it is still safe to turn on - but with limits in place to protect against content explosion.

This would be nice, but it gets difficult with HTML, since there's never a good way to truncate content and still leave it usable in any form.

> I believe an instance-wide flag (via a django setting) should be sufficient for this. We may not enable it on edx.org, for instance, unless there's a need for it.

The Course Blocks API's `student_view_data` query parameter allows you to [specify a list of XBlocks](http://edx.readthedocs.io/projects/edx-platform-api/en/latest/courses/blocks.html#query-parameters) whose `student_view_data` should be returned.

Does this do enough to prevent bloated API responses?

Reply:
@nasthagiri  As a follow-up to my comment above, I found when testing the Course Blocks API's `student_view_data` parameter that it wasn't working as expected.  Basically, if you provided any value, like `?student_view_data=anything`, then all the blocks would return `student_view_data` that had this method implemented.

I located the issue and fixed it with 1692d04.  Is it ok to include this fix in this PR, or should I create a separate one?   CC @bradenmacdonald 

Verification steps are in the PR description.

Reply:
@pomegranited yes, you can go ahead and include that fix as part of this PR.  I suggest keeping it as a separate commit, however, since it's logically a different change.  Thanks.

Reply:
> The Course Blocks API's student_view_data query parameter allows you to specify a list of XBlocks whose student_view_data should be returned. Does this do enough to prevent bloated API responses?

The concern is not so much at the time of responding to the API request.  The concern is a bloat in data that is collected (during the [collect phase of a Block Transformer](https://github.com/edx/edx-platform/blob/b7011ec85cd03be15c066458664be3a545d64255/openedx/core/djangoapps/content/block_structure/transformer.py#L74)).  This collected data is stored in S3 and cached in Memcached, which has a maximum size of 1MB at the moment.

By having a global flag, we can turn it off (if needed) without needing to rollback all the code changes.  For the future, @ormsbee has talked about possibly adding limits to block data size.

Reply:
@nasthagiri  Ah, that makes sense, thank you!  I'll ensure the flag is added and tested.